### PR TITLE
cw - added cowHealth field to UserCommons.java

### PR DESCRIPTION
**Pull request by cyril-wang**
In this PR, we added in the cowHealth field into `UserCommons.java` that keeps track of the average cow health in a user's common. In addition, we also implemented changes into the `UserCommonsController.java` that updated the average cow health when cows were bought or sold. 

To test, I recommend running `mvn test` first, then either running it locally on localhost or through the heroku-qa. While doing so, check that the cowHealth field shows up in the Play Page under `Your Farm Stats`. For example, like these screenshots. 
image
image

If you run into errors, such as errors with communicating with the backend or sql unable to perform query, when trying to join a common or when accessing the play page and leaderboard page, then try running `mvn clean` if on localhost or resetting the database if on heroku.

### CODE REVIEW

**Review by Geivet — APPROVED** [review]
LGTM!

**Review by nick-pellegrin — DISMISSED** [review]
Looks good!

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java [hunk 1]
```diff
@@ -189,6 +189,7 @@ public ResponseEntity<String> joinCommon(
         .username(username)
         .totalWealth(joinedCommons.getStartingBalance())
         .numOfCows(0)
+        .cowHealth(0)
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java:192]: Can we set the initial cow health to 100 instead?

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java [hunk 2]
```diff
@@ -89,6 +89,7 @@ public ResponseEntity<String> putUserCommonsByIdBuy(
 
         if(userCommons.getTotalWealth() >= commons.getCowPrice() ){
           userCommons.setTotalWealth(userCommons.getTotalWealth() - commons.getCowPrice());
+          userCommons.setCowHealth((userCommons.getCowHealth()*userCommons.getNumOfCows() + 100) / (userCommons.getNumOfCows() + 1));
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java:92]: No... we don't want to do this.  Instead, there is a job that will update cow health periodically.

There may be some old documentation that suggests this is the way to do it, but that's no longer how we are doing it.   Please leave cow health unchanged when we buy/sell cows.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java [hunk 3]
```diff
@@ -118,6 +119,9 @@ public ResponseEntity<String> putUserCommonsByIdSell(
         if(userCommons.getNumOfCows() >= 1 ){
           userCommons.setTotalWealth(userCommons.getTotalWealth() + commons.getCowPrice());
           userCommons.setNumOfCows(userCommons.getNumOfCows() - 1);
+          if (userCommons.getNumOfCows() == 0) {
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsController.java:122]: No... please leave the cow health unchanged when buying and selling cows.

**Review by pconrad — CHANGES_REQUESTED** [review]
There a few small changes I'd like to request based on an updated understanding of how cow health will work.

Otherwise, looks good to me.

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**